### ISSUE TITLE
upgrade 3.0.0-beta.2 to 3.0.0-beta.3 some file failed

### ISSUE THREAD

**Issue by MoonSeeing** [marmelab/react-admin#3924]
**What you were expecting:**
upgrade react-admin to  3.0.0-beta.3 from 2

**What happened instead:**
some file have problem :convertLegacyDataProvider.ts  

![image](https://user-images.githubusercontent.com/19420622/68122044-62aee980-ff44-11e9-8c93-1faf7c83f0a6.png)
**Steps to reproduce:**
yarn upgrade react-admin@3.0.0-beta.3
**Related code:**
```
import {
    CREATE,
    DELETE,
    DELETE_MANY,
    GET_LIST,
    GET_MANY,
    GET_MANY_REFERENCE,
    GET_ONE,
    UPDATE,
    UPDATE_MANY,
} from '../core';
```
../core did not have types should be 
**Environment**

* React-admin version:3.0.0-beta.2

**Comment by Kmaschta** [comment]
Thank you for opening this issue. In order to confirm it, we'll have to reproduce it.
As explained in the bug report template, please fork the following CodeSandbox and repeat your issue on it:

https://codesandbox.io/s/github/marmelab/react-admin/tree/next/examples/simple

This is the simplest way to confirm a bug is related to the React Admin codebase.

**Comment by fzaninotto** [comment]
What does "../core did not have types should be" mean? When you mean "some files fail", what does actually fail, and what's the error message?

**Comment by tdeo** [comment]
I'm running into a similar issue when using a "legacy" dataProvider, it seems that the bug was introduced in https://github.com/marmelab/react-admin/commit/dccc3cb1368f84fac9d2fb3f8e221f318afbf6b1, and since then this import is not resolving properly https://github.com/marmelab/react-admin/commit/dccc3cb1368f84fac9d2fb3f8e221f318afbf6b1#diff-d2b6c4d5acf1bd4486f924173f33fc4cR11

I also noticed that this for me only happens in `react-scripts` dev mode and not once build.

Here is a codesandbox illustrating the issue: https://codesandbox.io/s/optimistic-frog-mr8lf

**Comment by fzaninotto** [comment]
Thanks, I managed to reproduce the issue. I'm Investigating.

**Comment by fzaninotto** [comment]
fixed by #3927

**Comment by MoonSeeing** [comment]
> I'm running into a similar issue when using a "legacy" dataProvider, it seems that the bug was introduced in [dccc3cb](https://github.com/marmelab/react-admin/commit/dccc3cb1368f84fac9d2fb3f8e221f318afbf6b1), and since then this import is not resolving properly [dccc3cb#diff-d2b6c4d5acf1bd4486f924173f33fc4cR11](https://github.com/marmelab/react-admin/commit/dccc3cb1368f84fac9d2fb3f8e221f318afbf6b1#diff-d2b6c4d5acf1bd4486f924173f33fc4cR11)
> 
> I also noticed that this for me only happens in `react-scripts` dev mode and not once build.
> 
> Here is a codesandbox illustrating the issue: https://codesandbox.io/s/optimistic-frog-mr8lf

tdeo thank you for reproduce it ,same error with me .

**Comment by MoonSeeing** [comment]
> What does "../core did not have types should be" mean? When you mean "some files fail", what does actually fail, and what's the error message? also, you mention 2 separate versions in your description, which one are you using?

@fzaninotto  sorry , i did not  provide a clear description.anyway ,thank your attention.

**Comment by MoonSeeing** [comment]
> fixed by #3927
have you publish your fix @fzaninotto 
can  i upgrade package to beta.5 now? 
thanks!

**Comment by fzaninotto** [comment]
no, you'll have to wait until it's published.
